### ISSUE TITLE
Sass won't output compressed stylesheets

### ISSUE THREAD

**Issue by maxlapides** [bdkjones/CodeKit2#100]
All of the other Sass output styles seem to work correctly except Compressed. All of the line breaks remain. This occurs on all of my projects, including the one I zipped up for you for this issue: https://github.com/bdkjones/codekit2beta/issues/98

I'm on Mavericks 10.9.1 and I use Coda 2 (but this issue is independent of the text editor I use). This was also an issue in beta 3.

**Comment by bdkjones** [comment]
Is this a Compass project? Are you using libsass or Ruby Sass? Details matter.

**Comment by maxlapides** [comment]
Sorry! No, it's not a Compass project and I'm using Ruby Sass. (Actually, I've never gotten libsass to work at all - it just crashes CodeKit immediately.)

**Comment by maxlapides** [comment]
I just tested it and the compressed output works properly with Compass projects that are set with `output_style = :compressed`

**Comment by bdkjones** [comment]
Please zip a test project and put it online so I can download it. 

Please show me the Sass file with the "correct" compressed output and the Sass file with the incorrect output.

**Comment by maxlapides** [comment]
Here's one of my projects with the issue where Sass won't give me a compressed output: http://maxlapides.com/downloads/maxlapides.zip

If you open up that project, my Sass file are at wp-content/themes/maxlapides/library/scss.

The only case where I've gotten the compressed stylesheets to work is with a Compass project.

**Comment by bdkjones** [comment]
I downloaded that project, found "style.scss" and set its output style to compressed. Clicked "compile" and I got compressed CSS, just as expected. 

Please make sure you're using Beta 4. Then post a screenshot here of the inspector panel when you have "style.scss" selected in CodeKit.

**Comment by bdkjones** [comment]
I think I know what's happening. 

You've got two "style.css" files in this project. One in the library folder and one in library/css. Your Scss file is compiling its output to the latter path. I think you're viewing the former path, which CodeKit is not creating.

**Comment by maxlapides** [comment]
The style.css file in my theme folder is required by WordPress - it's the "theme stylesheet" (more info here: http://codex.wordpress.org/Theme_Development#Theme_Stylesheet). It should just sit there and never be touched.

Here's a link to what the compiled CSS file looks like for me: http://maxlapides.com/downloads/style.css.zip

Also, here's the screenshot of CodeKit that you requested: 

![2013-12-27_18-49-04](https://f.cloud.github.com/assets/1874682/1816808/78f1e8de-6f51-11e3-9781-ce16a5ee69dd.png)

Oh wait! Just realized that if I uncheck "Run Autoprefixer on the CSS file," it compresses the CSS properly! But then when I re-checked it, back to the uncompressed output :(

**Comment by bdkjones** [comment]
Yep. Autoprefixer is currently too dumb to minify CSS. 

I've opened a request for it on their page: https://github.com/ai/autoprefixer/issues/159

**Comment by bdkjones** [comment]
Oh, good news! Autoprefixer 1.0 fixes this. 

I've updated to Autoprefixer 1.0 in Beta 5, which I will post later tonight. Issue resolved.

**Comment by maxlapides** [comment]
Sweet! Thanks!

**Comment by ehausen** [comment]
For my the Output Style does nothing. I had to manually edit the config.rb file and set:
output_style = :compressed

Even commenting out that line and settings Output Style to compressed in Codekit won't help.

**Comment by bdkjones** [comment]
You have to make changes to the file before you save. Compass caches the file partway through the compiling process to deal with Ruby's atrociously awful performance. If you don't make any changes to the contents of the file, Compass does not recompile it, even if you alter settings in CodeKit.

There's nothing I can do about that.

**Comment by ehausen** [comment]
Maybe a tooltip somewhere on the settings page to avoid headaches and support request? :)

**Comment by fahimshekaib** [comment]
That's right! For my workflow I set sassStyle = 'compressed'; But it did not compress it until I made a minor change in one of the sass files. It is just the first time we need to do this. Afterwards, since we would already have a compressed "Production version CSS" file, recompilation would only be needed if indeed a change is made in any of the sass files. 
FYI:
I am using Ruby SASS, Compass, etc with Gulp.
